core/extension/gdextension_interface.cpp

```diff
@@ -28,7 +28,7 @@
 /* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
 /**************************************************************************/
 
-#include "gdextension_interface.h"
+#include "gdextension_interface.h" // encapsulation pragma: ignore // FIXME: legacy include.
```

Comment:
Why is it triggering? It's the header in the same folder, so it should be fine.

Reply:
I'm guessing the implemntation currently can't handle relative paths.
When I had this problem with the build profiler tool, I handled it by [asking clang's preprocessor for the full path](https://github.com/Ivorforce/clang-project-profiler/blob/main/main.cpp#L30-L33).

In this case, it might be enough to try looking for the file in parent folders progressively?

Reply:
Oh yeah, that's a false positive. Must've detected the file existing in the repository root after GDExtension generation

In that case, I'll go ahead and expand the implementation to handle relative paths too. It'd effectively be an `#include` validator by that point, so maybe the scope of this tool could be expanded too? Maybe not for this PR specifically, but I'll change the script name to something broader than `validate_encapsulation`